### PR TITLE
add address lookup with BC Geocoder to map search

### PR DESCRIPTION
**Pull request by mtCarto**
* [NRPT-177](https://bcmines.atlassian.net/browse/NRPT-177) add geocoder to map search

### CODE REVIEW

File: src/app/map/search/search.component.ts [hunk 1]
```diff
@@ -24,24 +26,28 @@ export class SearchComponent implements OnInit, OnChanges, OnDestroy {
...
   @Output() updateMatching = new EventEmitter();
+  @Output() showPlace = new EventEmitter();
 
   public loading = false;
   public ranSearch = false;
   public mineFilter: string = null;
   public _mineFilter: string = null; // temporary filters for Cancel feature
   public permitFilter: string = null;
-  public _permitFilter: string = null; // temporary filters for Cancel feaure
+  public _permitFilter: string = null; // temporary filters for Cancel feaure     is this necessary?
```
Comment (dhlevi) [inline src/app/map/search/search.component.ts:36]: seems like the _***Filter variables are all not really needed. In the end they just get copied over to the non-underscored version anyway in the `applyFilters` function and elsewhere?

`this.permitFilter = this._permitFilter;`

It probably makes sense at some point to just use a single var. Not really super important to change it all now though, but yeah, probably not necessary!

File: src/app/services/geocoder-api.ts [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+  // search within an area (bounding box)
+  private _bbox: string;  // A bounding box (xmin,ymin,xmax,ymax) that limits the search area
+
+  // search within an area (centre point + distance in meters)
+  private _centre: string;  // The coordinates of a centre point (x,y) used to define a bounding circle that will limit the search area
+  private _maxDistance: number;
+
+  constructor() {
+    this._baseUrl = 'https://geocoder.api.gov.bc.ca';
```
Comment (dhlevi) [inline src/app/services/geocoder-api.ts:24]: Not sure if we want these kind of hardcoded configs in a const file or an environment param?

Comment (mtCarto) [reply src/app/services/geocoder-api.ts:24]: Would be nice to have as an environment param, something like the solution EPIC had which would be a bit more work. The current api url's are hardcorded in a switch statement as well. A constants file would be the easiest to push in now

**Comment by mtCarto** [comment]
The default popup opening on highest score I wasn't 100% sure on, the popup often hides any other nearby results and to get a popup open for a clustered point was a bit fiddly. I opted for zooming in to a level that would force open the cluster containing the popup, which makes it harder to see nearby mines. On the other hand not opening the popup forces the user to click through the markers

**Comment by mtCarto** [comment]
The current popup behavior will be kept as is, demoed it as is to Jessica and she's happy with the current state.

File: src/app/services/geocoder-api.ts (hunk 2 above)
Comment (mtCarto) [reply src/app/services/geocoder-api.ts]: Added a shared constants file for the api urls, (geocoder and nrpti)

**Pull request merged**